server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -347,6 +355,32 @@ public static Item forInsert(String id,
             this.autoGeneratedTimestamp = autoGeneratedTimestamp;
         }
 
+        @Override
+        public long ramBytesUsed() {
+            if (usedBytes > -1) {
+                return usedBytes;
+            }
+            usedBytes = super.ramBytesUsed()
+                + Long.BYTES; // autoGeneratedTimestamp
+            if (source != null) {
+                usedBytes += source.length();
+            }
+            if (updateAssignments != null) {
+                for (var assignment : updateAssignments) {
+                    usedBytes += assignment.ramBytesUsed();
+                }
+            }
+            if (insertValues != null) {
+                for (var value : insertValues) {
+                    usedBytes += RamUsageEstimator.sizeOfObject(value);
+                }
+            }
+            for (String pkValue : pkValues) {
```

Comment:
Just double checking, I saw for streaming we return `List.of()`, but for other cases it's always empty list and not null?

Reply:
As far as I can tell it is never `null`